subscriber.rb

```diff
@@ -0,0 +1,48 @@
+
+require 'aws-sdk-sqs'
+require 'byebug'
+require_relative 'load_aws'
+
+class Subscriber
+  
+  ACCESS_KEY_ID = AWS["access_key_id"]
+  SECRET_ACCESS_KEY = AWS["secret_access_key"]
+  REGION = AWS["region"]
+
+  attr_reader :sqs
+
+  def initialize
+    creds = Aws::Credentials.new(ACCESS_KEY_ID, SECRET_ACCESS_KEY)
+    @sqs = Aws::SQS::Client.new(region: REGION, credentials: creds)
+  end
+
+# Todo: URL validation
+  def read_queue(options=[])
+    queue_url = options[0]
+    max_number_of_messages = options[1].nil? ? 10 : options[1]
+    wait_time_seconds = options[2].nil? ? 10 : options[2]
```

Comment:
What is `wait_time_seconds` ?

Reply:
wait_time_seconds is for long_polling. So it will wait for the specified amount of time before it polls again for messages.

> Amazon SQS will wait until a message is available in a queue before sending a response. Unless the connection times out, the response to the ReceiveMessage request contains at least one of the available messages, up to the maximum number of messages specified in the ReceiveMessage action.

Reply:
That is for long polling - don't believe we require long polling - just poll, and return 1 message from the queue. Sleep 5 seconds (or whatever that is configured) and then repeat the cycle.

Reply:
Even if we use short-polling (AWS defaults to short polling), it polls for a minimum of 30 seconds within which it receives all messages waiting in the queue. so with one poll we can actually pull in a lot more messages than just one. Even so in that case, max number of messages in a poll will not be exceeded and since the queue may be receiving messages continually, we can keep the receive message statement in the infinite loop. Its just that, it would have received multiple messages instead of one in each poll.

Reply:
Lets not pull in more messages than one at a time, process it and then sleep for a couple of seconds..

Reply:
If we are receiving only one message then it will process only one message at a time.. the loop is because 'sqs.receive_message' will always return an array of all the messages.. in our case its one.